shared/src/data/GoodNumber.ts

```diff
@@ -149,6 +173,10 @@ export class GN {
     return a.lt(b) ? a : b;
   }
 
+  static areWithinNSigDigs(a: GN, b: GN, n: number): boolean {
```

Comment:
I know you're just taking the name from elsewhere, but I kinda want to take this opportunity to call it "eqUpToNSigDigs" or "doFirstNSigDigsMatch" -- I feel these more accurately describe the condition we're checking.

Reply:
Updated it to `firstNSigDigsMatch`